bsd/enclosure.pyx

```diff
@@ -77,19 +77,22 @@ cdef class Enclosure(object):
             if not objp:
                 raise MemoryError('calloc objp failed')
 
-            res = ioctl(self.enc_fd, ses.ENCIOC_GETELMMAP, <ses.caddr_t>objp)
-            if res < 0:
-                raise RuntimeError('ioctl failed to get enclosure element map')
-
             elm_info = <ses.elm_info_t*>calloc(num_elms, sizeof(ses.elm_info_t))
             if not elm_info:
+                self.clean_up(num_elms, elm_info, objp)
```

Comment:
Can a try/finally block be used instead?

Reply:
yep, done.